app/src/main/java/org/mozilla/reference/browser/HomeHandler.kt

```diff
@@ -0,0 +1,14 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package org.mozilla.reference.browser
+
+/**
+ * Interface for fragments that want to handle 'home' button presses.
+ */
+interface HomeHandler {
+    fun onHomePressed(): Boolean
+}
```

Comment:
nit: I wonder if `HomeHandler` can be misleading since according to the docs `onUserLeaveHint()` is not only about "home"?

Reply:
Fair point. How about 'UserInteractionHandler'? I've put more comments in the `onHomePressed` method as well for clarity.